### ISSUE TITLE
waf.bat and fake python.exe on Windows 10+

### ISSUE THREAD

**Issue by a1batross** [FWGS/xash3d-fwgs#854]
$subj

For some idiotic reason Microsoft puts fake python.exe into PATH on Windows 10 that should open Microsoft Store page of Python3 but knowing the reliability of Microsoft Store, it never works in the intended way, probably freezing the build, and we had such reports on in the past.

I suggest the removal of waf.bat. There isn't much of a problem to just add real Python to the PATH and call it directly, without the extra batch script.

**Comment by Velaron** [comment]
What about checking the path of the executable and printing a useful error to the user?

**Comment by Velaron** [comment]
My system gives a warning btw

"Python was not found; run without arguments to install from the Microsoft Store, or disable this shortcut from Settings > Manage App Execution Aliases."

**Comment by Velaron** [comment]
It also seems as the executable sets errorlevel to 9009 (as if file is not found), so maybe you could check that after executing the placeholder .exe?

**Comment by a1batross** [comment]
where.exe seems doesn't exist on old Windows.

We could check if it's not available then just launch python.

Or if it gives any output that it's not fake python.exe in PATH. :think:

ср, 25 мая 2022 г., 07:46 Bohdan Shulyar ***@***.***>:

**Comment by Velaron** [comment]
Yeah, that makes sense, as this problem is Windows 10 specific and where.exe exists in all installations this could work.
Example output when Python is installed:
`
PS C:\Users\Velaron> where.exe python
C:\Users\Velaron\AppData\Local\Programs\Python\Python310\python.exe
C:\Users\Velaron\AppData\Local\Programs\Python\Python39\python.exe
C:\Users\Velaron\AppData\Local\Microsoft\WindowsApps\python.exe
`
There is another problem, the user needs to add to the system PATH variable, as it takes over the user one, so Python could be installed on the system, but the placeholder fake one is first on the list, i.e.
`
C:\Users\Velaron\AppData\Local\Microsoft\WindowsApps\python.exe
C:\Users\Velaron\AppData\Local\Programs\Python\Python310\python.exe
C:\Users\Velaron\AppData\Local\Programs\Python\Python39\python.exe
`

So basically you need to check if where.exe exists, and then if the first entry doesn't point to WindowsApps folder.

**Comment by a1batross** [comment]
We also could use py.exe wrapper, that's included in official releases, as well.

Can you check if it does work there?

**Comment by Velaron** [comment]
`
PS C:\Users\Velaron> where.exe py
C:\Windows\py.exe
PS C:\Users\Velaron> py
Python 3.10.1 (tags/v3.10.1:2cd268a, Dec  6 2021, 19:10:37) [MSC v.1929 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
`

**Comment by Velaron** [comment]
Maybe I need to wind up a VM to check if the executable is pre-installed, if you mean that.

**Comment by a1batross** [comment]
No, don't need to. 

I'm just a bit busy, will check myself later.

**Comment by a1batross** [comment]
@Velaron 

Tried to fix that in https://github.com/FWGS/xash3d-fwgs/commit/060835cf2fadb8b4873005e3d4d8f8b2ba6ed550

It turns out, reg query wasn't parsed correctly on newer systems, so it fallback to just python.exe.

**Comment by Velaron** [comment]
Works perfectly now without adding python to PATH.

**Comment by a1batross** [comment]
Awesome, will propose fixes to upstream waf then
